src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -20,15 +20,26 @@ namespace Microsoft.CmdPal.UI;
 public sealed partial class ListPage : Page,
     IRecipient<NavigateNextCommand>,
     IRecipient<NavigatePreviousCommand>,
-    IRecipient<ActivateSelectedListItemMessage>,
-    IRecipient<ShowExceptionMessage>
+    IRecipient<ActivateSelectedListItemMessage>
 {
     private readonly DispatcherQueue _queue = DispatcherQueue.GetForCurrentThread();
 
     public ListViewModel? ViewModel
     {
         get => (ListViewModel?)GetValue(ViewModelProperty);
-        set => SetValue(ViewModelProperty, value);
+        set
+        {
+            if (ViewModel != null)
+            {
+                ViewModel.PropertyChanged -= ViewModel_PropertyChanged;
+            }
+
+            SetValue(ViewModelProperty, value);
+            if (value != null)
+            {
+                value.PropertyChanged += ViewModel_PropertyChanged;
+            }
```

Comment:
We should always try and avoid doing things in the property setters at all costs...

This is a DependencyProperty, so the registration already provides you a property changed callback via the `PropertyMetadata` constructor, you just need to pass in your method name as the 2nd argument there.

Your `ViewModel_PropertyChanged` will just become static and have different arguments, but do the same thing you need, see this example:

https://github.com/zadjii-msft/PowerToys/blob/44070f83bd0fb65aeba646d0dce7ec233ec150a0/src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs#L37-L50